blsSignatures/blsSignatures.go

```diff
@@ -43,9 +45,11 @@ func GenerateKeys() (PublicKey, PrivateKey, error) {
 }
 
 func PublicKeyFromPrivateKey(privateKey PrivateKey) (PublicKey, error) {
-	pubKey := &bls12381.PointG2{}
-	g2 := bls12381.NewG2()
-	g2.MulScalar(pubKey, g2.One(), privateKey)
+	pubKey := new(bls12381.G2Affine)
+	_, _, _, g2 := bls12381.Generators()
+	fr := new(fr.Element)
+	fr.Set(privateKey)
+	pubKey.ScalarMultiplication(&g2, fr.BigInt(new(big.Int)))
```

Comment:
I don't think we need to make a copy of privateKey here, as BigInt doesn't seem to mutate the passed in Element. I'd just do `privateKey.BigInt(new(big.Int))` instead of making the `fr` variable.